kore/src/Kore/Step/SMT/Representation/All.hs

```diff
@@ -46,9 +53,40 @@ sorts).
 build
     :: VerifiedModule Attribute.Symbol
     -> Map.Map Id Attribute.Constructors
-    -> AST.SmtDeclarations
+    -> Step.AST.SmtDeclarations
 build indexedModule sortConstructors =
-    resolve (sorts `AST.mergePreferFirst` symbols)
+    removeDuplicates $ resolve (sorts `Step.AST.mergePreferFirst` symbols)
   where
     sorts = Sorts.buildRepresentations indexedModule sortConstructors
     symbols = Symbols.buildRepresentations indexedModule
+
+removeDuplicates :: Step.AST.SmtDeclarations -> Step.AST.SmtDeclarations
+removeDuplicates Step.AST.Declarations { sorts, symbols } =
+    let constructorSymbols =
+            join $ mapMaybe getConstructorNames (Map.elems sorts)
+     in Step.AST.Declarations
+         { sorts
+         , symbols = filter (isNotDuplicate constructorSymbols) symbols
+         }
+  where
+    getConstructorNames
```

Comment:
We should make this a left-fold returning a `Set`:

```.hs
getConstructorNames
    :: Set Text
    -> Step.AST.Sort SMT.AST.SExpr Text Text
    -> Set Text
```

so that above we need only `let constructorSymbols = foldl' getConstructorNames Set.empty (Map.elems sorts)`. A `Set` will be more efficient for filtering symbols below.